guides/web-apps/acceptance-testing-with-selenium/README.md

```diff
@@ -114,157 +84,67 @@ We don't want to test only Firefox. Because of that, you should use Selenium Rem
-                break;
-            case "edge":
-                capabilities = DesiredCapabilities.Edge();
-                driver = new RemoteWebDriver(remoteWebDriverUri, capabilities);
-                break;
-            default:
-                capabilities = DesiredCapabilities.Firefox();
-                driver = new RemoteWebDriver(remoteWebDriverUri, capabilities);
-                break;
-        }
-        return driver;
-    }
-}
-```
+- Put the `jar` file as well as 3 `exe` files directly in `C:\Selenium`
 
-Now, we need to adapt the test file to make use of it.
+Open your <code>Properties</code> in the <code>Tests</code> project. Go to <code>Reference Paths</code>, enter C:\Selenium and click Add Folder.
 
-In the file you created in the previous step (`PageLoads.cs`), locate the following line:
+### Use BaseTest class to run tests in multiple browsers
 
-```cs
-driver = new FirefoxDriver();
-```
+BaseTest is a helper class that makes it easier to test multiple browsers. The source code is available:
 
-Change it to:
+- BaseTest helper class [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/BaseTest.cs)
+- Using of BaseTest class [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionBoolean/CheckboxPageTest.cs)
 
-```cs
-driver = WebDriverFactory.Create(this.browser);
-```
+When you rebuild the test project now, you should see each test for every browser.
 
-Now, what is `this.browser`? This is a fixture parameter that carries the name of the browser used to execute the test.
+The final setup looks like this:
 
-What this means is that you need to change every instance of :
+![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
 
-```cs
-[TestFixture]
-```
+Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.*.jar`.
 
-To be:
+## Wait for asynchronous content
 
-```cs
-[TestFixture("firefox")]
-[TestFixture("chrome")]
-[TestFixture("edge")]
-```
+There is one common pitfall when writing Selenium tests. The test is executed with disregard of the asynchronously loaded content. This means that your tests need to explicitly wait for UI elements to appear before running any actions or assertions on them.
 
-Finally, add this constructor somewhere on top of the class. It stores the fixture parameter in the `this.browser` variable:
+It is a good practice to always wait:
 
+- Wait for a text element to be present before you check the content of that element
 
-```cs
-private string browser;
+Example can be find [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs) 
 
-public PageLoads(string browser)
-{
-    this.browser = browser;
-}
-```
+Please take a look at TextareaPage_WriteToTextArea() test and line #37. 
```

Comment:
Please change TextareaPage_WriteToTextArea() to `[TextareaPage_WriteToTextArea()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L29)` 

"line #37" to `[line 37](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L37)`